Documentation/Formatter Usage Guide.md

```diff
@@ -0,0 +1,300 @@
+extension RxPokemonDetailsPresenter: RxPokemonDetailsPresenterInterface {
+
+    func configure(with output: RxPokemonDetails.ViewOutput) -> RxPokemonDetails.ViewInput {
+        let pokemon = Driver.just(pokemon)
+        let formatterInput = RxPokemonDetails.FormatterInput(models: pokemon)
+
+        let formatterOutput = formatter.format(for: formatterInput)
+
+        let title = Driver.just(self.pokemon.title).compactMap { $0 }
+        let imageURL = Signal.just(self.pokemon.imageURL).compactMap { $0 }
+        return RxPokemonDetails.ViewInput(models: formatterOutput, events: RxPokemonDetailsEvent(title: title, imageURL: imageURL))
+    }
+
+}
+```
+
+We added a property Pokemon since we open the screen by clicking on a Pokemon and we don't make any API calls. Since this is a simple example, we only have the <i>configure</i> function. Inside the configure function we configured our Formatter by calling the <i>format</i> function and our *ViewInput*. It's the same as we did it previously in the RxModule, we only have the Formatter component now.
+
+As you can see, the Presenter doesn't have any mapping logic of our models it only passes the models to the Formatter.
+
+### ViewController, Wireframe and Interactor
+
+There aren't any changes on how to use ViewController, Wireframe, and Interactor. We use the ViewController the same way as we did in the RxModule, by subscribing to events and reacting to them.
+
+### Formatter
+
+```swift
+final class RxPokemonDetailsFormatter {
+}
+
+// MARK: - Extensions -
+
+extension RxPokemonDetailsFormatter: RxPokemonDetailsFormatterInterface {
+
+    func format(for input: RxPokemonDetails.FormatterInput) -> RxPokemonDetails.FormatterOutput {
+        return RxPokemonDetails.FormatterOutput(sections: createTableViewSections(from: input.models))
+    }
+
+}
+
+private extension RxPokemonDetailsFormatter {
+
+    func createTableViewSections(from pokemon: Driver<Pokemon>) -> Driver<[TableSectionItem]> {
+        pokemon
+            .map { [unowned self] in  createSections(with: $0)}
+    }
+
+    func createSections(with pokemon: Pokemon) -> [TableSectionItem] {
+        var items: [TableSectionItem] = []
+        items.append(createDescriptionSection(pokemon))
+        return items
+    }
+
+    func createDescriptionSection(_ pokemon: Pokemon) -> TableSectionItem {
+        return RxPokemonDetailsSection(items: [RxPokemonDetailsItem(model: PokemonDetailsItem.description(pokemon))])
+    }
+
+}
+```
+
+Lastly, we get to our beloved Formatter. We only have the formatting/mapping logic inside the Formatter. Since it is a simple example, it doesn't have many lines of code but it still helps us with the maintainability and readability following the SRP. Inside the <i>format</i> function we use the same principle as with the Presenter. Take the items from the input and convert them into needed models. The Formatter can have much more formatting and item creation logic and that's why you want it in one place.
```

Comment:
```suggestion
Lastly, we get to our beloved Formatter. We only have the formatting/mapping logic inside the Formatter. Since it is a simple example, it doesn't have many lines of code but it still helps us with the maintainability and readability following the SRP. Inside the <i>format</i> function we use the same principle as with the Presenter. Formatter will take the networking models from the input and convert them into items consumed by the view. The Formatter can have much more formatting and item creation logic and that's why you want it in one place.
```